nixos/modules/services/misc/ollama.nix

```diff
@@ -87,7 +120,8 @@ in
         ExecStart = "${lib.getExe ollamaPackage} serve";
         WorkingDirectory = cfg.home;
         StateDirectory = [ "ollama" ];
-        DynamicUser = true;
+        DynamicUser = cfg.sandbox;
+        ReadWritePaths = cfg.writablePaths;
```

Comment:
This solves the problem with file system, but it seems like systemd also sandboxes away network. So you can't do, for example, `ollama pull`.

Reply:
I have tested the service's network functionality, and haven't had any problems getting it to download new models. Are you sure the problem is with the network? If it is, are you confident that it doesn't have anything to do with your local network configuration?

Could you try downloading a new model (`tinydolphin` is pretty small) with the default ollama service configuration (i.e. just `services.ollama.enable = true;`, and get back to me with the results of that?